### PR TITLE
Add config for internal school beacons dataset

### PR DESCRIPTION
**Pull request by chiaberry**
https://github.com/cityofaustin/atd-data-tech/issues/13946

https://datahub.austintexas.gov/Transportation-and-Mobility/School-Zone-Beacons-Internal/8whb-sg4d
it is an internal dataset, so you will need to login to see it

corresponding airflow PR: https://github.com/cityofaustin/atd-airflow/pull/181

I changed a column in the knack view to use the text version (ATD_location_id) of the column also named `location` that had this: `LOC23-022820` since it was throwing errors in socrata to have two different columns named location, where one was text and not a location type. 

<br class="Apple-interchange-newline">

### CODE REVIEW

File: services/records_to_knack.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def format_filter_date(date_from_args):
 
 
 def get_pks(field_map, app_name_dest):
-    """ return the src and destination field name of the primay key """
```
Comment (johnclary) [inline services/records_to_knack.py:18]: 🙏

File: services/config/knack.py [hunk 2]
```diff
@@ -176,6 +176,12 @@
             "layer_id": 0,
             "item_type": "layer",
         },
+        "view_3898": {
+            "description": "School Beacons EDP",
```
Comment (johnclary) [inline services/config/knack.py]: can there be a better description for this that makes it clear why it's different from the "School Beacons" entry? 

why can't these two views be combined? 

why is this an internal only dataset?

**Review by johnclary — CHANGES_REQUESTED** [review]

File: services/config/knack.py (hunk 2 above)
Comment (Charlie-Henry) [reply services/config/knack.py]: Looks like the dataset contains created by and modified by fields, that could be why?

Comment (chiaberry) [reply services/config/knack.py]: yes, its internal only because of PII. I dont know why they want those fields in there, I will ask Christina and Karo. There was an attempt to use the previous view, but for some reason they had to use a new one.

**Review by Charlie-Henry — APPROVED** [review]
Maybe call it School Beacons (private view)? 🚢

File: services/records_to_knack.py
Comment (chiaberry) [inline services/records_to_knack.py:1]: these changes have nothing to do with the school beacons dataset, and I am happy to move them into their own PR. 

At some point in the past I was debugging what I thought was a knack to knack error, and while looking at this script I added some comments to help me understand what was happening.

**Review by johnclary — APPROVED** [review]
thanks for making theknack2knack code easier to follow!

this looks good to me! 🚢
